### PR TITLE
Improve figures

### PR DESCRIPTION
**Pull request by henrikt-ma**
First of all: Due to recent confusion about when to use the _draft_ state of a pull request, I'm initiating this as a draft to reflect that the proposed changes aren't based on the agreed resolution of an issue.  Still, this is a non-draft pull request in the sense that there is nothing more I'm planning to add before this is ready for proper review and merging.

This PR combines a couple of changes related to figures:
- Make sure every figure has a caption below the figure content.
- Improved formatting of figure captions.
- Some graphics are improved for better layout, tighter integration with the text, and sharp (scalable) looks in the HTML build.

### CODE REVIEW

**Comment by henrikt-ma** [comment]
@sjoelund, this PR also contains an added LaTeX package dependency, and I was considering trying the Docker dance myself, but realized I didn't even have an account.  Now I have one (henriktma), but it isn't part of the organization.  Could you please help once more by either updating the image, or adding me to the organization so that I can try doing it myself?

**Comment by sjoelund** [comment]
I have added you with write access on hub.docker.com. I suggest having Dockerfile.incremental based on that same image from April and trying it yourself. Otherwise it's quite fast for me to do it if you don't have docker setup.

**Comment by HansOlsson** [comment]
Since it doesn't impact the semantics, and it's not _that_ major we can either discuss it at the next phone meeting or before that.

**Comment by henrikt-ma** [comment]
> Since it doesn't impact the semantics, and it's not _that_ major we can either discuss it at the next phone meeting or before that.

Sure, but how to make that discussion happen before the meeting?  Should I remove the _Draft_ state and send an email?

**Comment by henrikt-ma** [comment]
> > Since it doesn't impact the semantics, and it's not _that_ major we can either discuss it at the next phone meeting or before that.
> 
> Sure, but how to make that discussion happen before the meeting? Should I remove the _Draft_ state and send an email?

I take the silence as an approval of proceeding as suggested.

**Comment by henrikt-ma** [comment]
Figures are slowly coming into shape, but there is one that is particularly challenging to me, namely _image2.png_, the collage of several model diagrams and one `MultiBody` visualization.  I don't have access to a SVG capable Modelica tool myself, but perhaps someone else would like to contribute with SVG images of the model diagrams (and possibly also of the `MultiBody` visualization)?  I don't know the level of SVG support in different tools, but I've heard that at least OPTIMICA Compiler Toolkit is good in this regard, right @jespermattsson (couldn't find the account of Axel Mårtensson)?

**Comment by sjoelund** [comment]
OMEdit will export model diagrams to SVG. GitHub doesn't like them in tickets though... Which model diagrams are needed?
[DoublePendulum.svg.gz](https://github.com/modelica/ModelicaSpecification/files/5167932/DoublePendulum.svg.gz)

**Comment by HansOlsson** [comment]
> OMEdit will export model diagrams to SVG. GitHub doesn't like them in tickets though... Which model diagrams are needed?
> [DoublePendulum.svg.gz](https://github.com/modelica/ModelicaSpecification/files/5167932/DoublePendulum.svg.gz)

It was one image with diagrams of various models from the various domains.
Dymola can also export diagrams as SVG, if we just can figure out what the models are.

And one image for the double-pendulum animation. 

Dymola can currently not export that as SVG; and I am not sure if any tool can do it. I see two possibilities:

- Make a high-res png and convert to svg (this seems possible)
- Export as VRML and figure out if there is a VRML->svg converter (couldn't find any)
Luckily it's just double-pendulum - so fairly simple.

**Comment by sjoelund** [comment]
> * Make a high-res png and convert to svg (this seems possible)

Some methods may be better than others. You often end up with gaps between the lines that should not be there. Maybe an 8K screen taking a really high-res PNG would suffice if you have a good software like Illustrator.

> * Export as VRML and figure out if there is a VRML->svg converter (couldn't find any)
>   Luckily it's just double-pendulum - so fairly simple.

I tried FreeCAD as it seemed to have SVG export. But it seems that is only for its 2D mode, and when printing to PDF it will rasterize the image. And converting to x3d to run it in Chrome also only gives a PNG.

**Comment by henrikt-ma** [comment]
> Dymola can currently not export that as SVG; and I am not sure if any tool can do it. I see two possibilities:
> 
> * Make a high-res png and convert to svg (this seems possible)

That would just be confusing, in my opinion; better keep it as a raster image in that case.  That one of the sub-images of the collage is a raster image doesn't mean there is no point having scalable graphics for the others.

**Comment by henrikt-ma** [comment]
Another figure where I could use some help from an SVG capable tool is the plot of the state machine behavior.  Any volunteers?

**Comment by sjoelund** [comment]
There is no nice + sign in statemachineplot from OMEdit to show when the clock ticks, but the results seem to be the same...

[statemachineplot.csv.gz](https://github.com/modelica/ModelicaSpecification/files/5181298/statemachineplot.csv.gz)
[statemachineplot.svg.gz](https://github.com/modelica/ModelicaSpecification/files/5181300/statemachineplot.svg.gz)

**Comment by henrikt-ma** [comment]
> There is no nice + sign in statemachineplot from OMEdit to show when the clock ticks, but the results seem to be the same...
> 
> [statemachineplot.csv.gz](https://github.com/modelica/ModelicaSpecification/files/5181298/statemachineplot.csv.gz)
> [statemachineplot.svg.gz](https://github.com/modelica/ModelicaSpecification/files/5181300/statemachineplot.svg.gz)

Great!  Regarding the + sign, I suppose it would make sense to align the looks of this figure with how a clocked variable is presented in section 16.2.1 _Clocks and Clocked Variables_, which will be easy now that you provided the data.

**Comment by henrikt-ma** [comment]
With the only thing remaining being that collage, I think it is time to get this merged and save the collage for a rainy day.

**Comment by HansOlsson** [comment]
I noticed some odd warnings from latexml about the size of the svg-images, but I don't see any problem with the result. Is that harmless?

**Comment by HansOlsson** [comment]
Oh, and I agree that it should be merged if the above is resolved.
After rebasing on master, of course. (I believe it's best you rebase @henrikt-ma  - as it seems most of the changes are with your other changes.)

**Comment by henrikt-ma** [comment]
> I noticed some odd warnings from latexml about the size of the svg-images, but I don't see any problem with the result. Is that harmless?

I don't get these warnings when building locally, probably because I have ImageMagick installed.  For convenience, I'm including an example of one of the warnings seen on Jenkins here:
> (Graphics MLS.html processing...
> Warning:imageprocessing:imagesize No image processing module found for image sizing
>	Will omit image size for /var/lib/jenkins1/ws/on_ModelicaSpecification_PR-2609/media/Modelica_Language.svg
>	Please install one of: Graphics::Magick,Image::Magick
>	In Post::Graphics[@0x55c96d5b8868] ->transformGraphic
> 0.01 sec)

When I look at the result, images seem to have their intrinsic size just like when I build locally without seeing the warnings.  Hence, I think we can consider the warnings harmless.

File: chapters/packages.tex [hunk 1]
```diff
@@ -195,6 +195,12 @@ \subsection{Example of Searching MODELICAPATH}\label{example-of-searching-modeli
 hierarchy to the file system.  The subpackage \lstinline!Pack2! is stored in its own subdirectory or file in the subdirectory \filename{MyLib}.  In this case the search
 succeeds and the package \lstinline!MyLib.Pack2! is loaded into the environment.
 \end{example}
+
+
+If during lookup a top-level name is not found in the unnamed top-level
+scope, the search continues in the package hierarchies stored in these
+directories.
```
Comment (HansOlsson) [inline chapters/packages.tex:202]: This paragraph is now duplicated. It was part of a section that moved in master-branch.

Comment (henrikt-ma) [reply chapters/packages.tex]: Well spotted!  Fixed.  Thanks!

**Pull request merged**